util/azure/azure.go

```diff
@@ -27,6 +29,9 @@ import (
 	azureautorest "github.com/Azure/go-autorest/autorest/azure"
 	"github.com/Azure/go-autorest/autorest/azure/auth"
 	"github.com/jongio/azidext/go/azidext"
+	"github.com/pkg/errors"
+	expv1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
```

Comment:
how does this not create an import cycle? since exp/api/v1beta1 imports this to use the `FindParentMachinePoolWithRetry` func

Reply:
This is because it's importing from CAPI's exp/api/v1beta1 since we only need to find the parent `MachinePool`. For `GetSubscriptionID`, we need access to the `AzureMachine` type from CAPZ.